ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -103,7 +163,10 @@ function TokenList({ onTokenClick, safeChains }: TokenListProps) {
       return;
     }
 
-    onTokenClick(token.chainId, token.address);
+    const tokenAddress =
+      isEvmChainId(token.chainId) && token.isNative ? '' : token.address;
```

Comment:
The route requires evm native tokens to not pass the token address, but at the same time it requires non-evm native tokens to pass it.